state/src/impls/shard_level.rs

```diff
@@ -241,6 +243,39 @@ impl<'db> ShardLevelState<'db> {
         Ok(())
     }
 
+    fn check_origin_outputs_of_orders(&self, inputs: &[AssetTransferInput]) -> StateResult<()> {
+        let mut origin_outputs_counter = HashMap::new();
+        for input in inputs {
+            let transaction_hash = input.prev_out.transaction_hash;
+            let index = input.prev_out.index;
+            let address = OwnedAssetAddress::new(transaction_hash, index, self.shard_id);
+            let asset = self
+                .asset(&address)
+                .unwrap()
+                .expect("It should exists because it's checked in check_and_run_input_script");
+
+            if asset.order().is_some() {
+                continue
```

Comment:
Is it okay to not check whether Order of the asset is the same with Order of the input?